datacapture/src/test/java/com/google/android/fhir/datacapture/validation/RegexValidatorTest.kt

```diff
@@ -0,0 +1,207 @@
+    checkAnswerNotMatchRegex(regex = "[0-9]+", value = IntegerType(-1))
+  }
+
+  @Test
+  fun int_matchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(regex = "[0-9]+", value = IntegerType(1234567890))
+  }
+
+  @Test
+  fun dateType_notMatchRegex_shouldReturnInvalidResult() {
+    val dateFormat = SimpleDateFormat("yyyy-MM-dd")
+    checkAnswerNotMatchRegex(
+      regex = "[0-9]{2}-[0-9]{2}-[0-9]{2}",
+      value = DateType(dateFormat.parse("2021-06-01"))
+    )
+  }
+
+  @Test
+  fun date_matchRegex_shouldReturnValidResult() {
+    val dateFormat = SimpleDateFormat("yyyy-MM-dd")
+    checkAnswerMatchRegex(
+      regex = "[0-9]{4}-[0-9]{2}-[0-9]{2}",
+      value = DateType(dateFormat.parse("2021-06-01"))
+    )
+  }
+
+  @Test
+  fun time_matchRegex_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(regex = "[0-9]{2}:[0-9]{2}", value = TimeType("18:00:59"))
+  }
+
+  @Test
+  fun time_notMatchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(regex = "[0-9]{2}:[0-9]{2}:[0-9]{2}", value = TimeType("18:00:59"))
+  }
+
+  @Test
+  fun string_notMatch_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(
+      regex = "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]",
+      value = StringType("www.hl7.org")
+    )
+  }
+
+  @Test
+  fun string_matchRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex(
+      regex = "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]",
+      value = StringType("https://www.hl7.org/")
+    )
+  }
+
+  @Test
+  fun uri_notMatch_shouldReturnInvalidResult() {
+    checkAnswerNotMatchRegex(regex = "[a-z]+", value = UriType(URI.create("https://www.hl7.org/")))
+  }
+
+  @Test
+  fun uri_answerOverMinLength_shouldReturnValidResult() {
+    checkAnswerMatchRegex(
+      regex = "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]",
+      value = UriType(URI.create("https://www.hl7.org/"))
+    )
+  }
+
+  @Test
+  fun invalidRegex_shouldReturnValidResult() {
+    checkAnswerMatchRegex("[.*", StringType("http://www.google.com"))
+  }
+
+  @Test
+  fun nonPrimitive_notMatch_shouldReturnValidResult() {
+    val requirement =
+      Questionnaire.QuestionnaireItemComponent().apply {
+        addExtension(
+          Extension().apply {
+            url = REGEX_EXTENSION_URL
+            this.setValue(StringType("[0-9]+"))
+          }
+        )
+      }
+    val response =
+      QuestionnaireResponse.QuestionnaireResponseItemComponent().apply {
+        addAnswer(
+          QuestionnaireResponseItemAnswerComponent().apply { this.value = Quantity(1234567.89) }
+        )
+      }
+
+    val validationResult = RegexValidator.validate(requirement, response)
+
+    assertThat(validationResult.isValid).isTrue()
+    assertThat(validationResult.message.isNullOrBlank()).isTrue()
+  }
+
+  private companion object {
+
+    @JvmStatic
+    fun checkAnswerMatchRegex(regex: String, value: PrimitiveType<*>) {
```

Comment:
super nit and optional: 
```suggestion
    fun checkAnswerMatchingRegex(regex: String, value: PrimitiveType<*>) {
```